### ISSUE TITLE
Make it possible to override the backend in a stack settings file

### ISSUE THREAD

**Issue by bob-bins** [pulumi/pulumi#10039]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Scenario: I have a Pulumi Project that has a Stack per environment. I am using the S3 backend and I want each environment's backend to be a different S3 bucket. I want Pulumi to use the correct backend just by running `pulumi up --stack <stackName>` (without having to re-run `pulumi login`).

Currently it's only possible to specify a backend per project, not per stack. This forces me to either use a single backend/S3 bucket for all environments or to run `pulumi login` every time before switching stacks.

**Comment by Frassle** [comment]
Is this to try and work around https://github.com/pulumi/pulumi/issues/2814?

**Comment by bob-bins** [comment]
Not quite. This ticket is asking to be able to set backends at the Stack level rather than be restricted to the same backend for the entire Project with IaC, which I don't see a mention of or reference to in #2814

**Comment by benlongo** [comment]
For instance, if I have prod and dev stacks, I might want those to go into two distinct S3 buckets.

**Comment by sam-utila** [comment]
Any progress on that?

I want to avoid a `pulumi login` and have a default backend for the entire project + set an override backend for specific prod stacks which I do have configuration files for committed in the source control

**Comment by sam-utila** [comment]
Could also contribute a PR in the CLI that does that, doesn't seem too hard

**Comment by Frassle** [comment]
I suspect this is surprisingly tricky. There's quite a lot of assumptions about the order things happen in the system. I think you'd have to do a partial load of the stack config to see if a "Backend" value is set (because a full config load needs the backend for ESC operation). Then after that actually do a proper config load. Given all the place config and stacks are loaded this is probably a fairly big change to make.
